src/java/org/apache/cassandra/cql3/QueryEvents.java

```diff
@@ -84,12 +87,25 @@ public void notifyQuerySuccess(CQLStatement statement,
         }
     }
 
+    private void updateMetrics(CQLStatement statement, Exception cause)
+    {
+        if (statement instanceof CQLStatement.SingleKeyspaceCqlStatement)
```

Comment:
What about CQL statements with no keyspace specified? Or where the reason it's an IRE is the keyspace name was wrong?  Will those be captured somewhere else?

Reply:
I think they will be captured in `ClientMetrics.unknownException` with all the other exceptions (with keyspace or not).
If we don't have a keyspace, I don't think we can identify a tenant ?